File: python/OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART/OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART/class_adapter/AdaptadorCReservacionMultiple.py
from OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART.GestorReservaciones import GestorReservaciones
from OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART.class_adapter.IReservacionMultiple import IReservacionMultiple
from OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART.whole_part.ReservacionVIP import ReservacionVIP
from OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART.whole_part.Reservacion import Reservacion
        
class AdaptadorCReservacionMultiple(GestorReservaciones,IReservacionMultiple):
    
    def __init__(self):
        self.reservaciones = []
    
    def crearReservacion(self, datosReservaciones):
        for datosReservacion in datosReservaciones:
            if len(datosReservacion) == 10:
                self.agregarReservacion(Reservacion(*datosReservacion))
            else:
                self.agregarReservacion(ReservacionVIP(*datosReservacion))
        return self.getReservaciones()
        
    
    def agregarReservacion(self, reservacion):
        self.reservaciones.append(reservacion)
    
    def getReservaciones(self):
        return self.reservaciones

File: php/OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART/claseAdaptador/AdaptadorCReservacionMultiple.php
<?php

require_once 'IReservacionMultiple.php';
require_once './principal/GestorReservacion.php';
require_once './datos/Reservacion.php';
require_once './datos/ReservacionVIP.php';



class AdaptadorCReservacionMultiple extends GestorReservacion implements IReservacionMultiple{
    
    public $reservaciones;
    
    public function __construct() {
        parent::__construct();
        
        $this->reservaciones = array();
    }
    
    public function crearReservacion($datosReservaciones) {
        foreach ($datosReservaciones as $datos) {
            if (count($datos) == 13){
                $r = new Reservacion($datos["nombre"], $datos["apellidoP"], $datos["apellidoM"], $datos["edad"], 
                                    $datos["tipoCliente"], $datos["edad"], $datos["dia"], $datos["mes"], $datos["anio"],
                                    $datos["hora"], $datos["pelicula"], $datos["numTarjeta"], $datos["fecha"]);
                $this->agregarReservacion($r);
            }else{
                $r = new ReservacionVIP($datos["nombre"], $datos["apellidoP"], $datos["apellidoM"], $datos["edad"], 
                                        $datos["tipoCliente"], $datos["edad"], $datos["dia"], $datos["mes"], $datos["anio"],
                                        $datos["hora"], $datos["pelicula"], $datos["numTarjeta"], $datos["fecha"], $datos["beneficios"]);
                $this->agregarReservacion($r);
            }
        }
        return $this->getReservaciones();
    }
    
    public function agregarReservacion($reservacion) {
        array_push($this->reservaciones, $reservacion);
    }

    public function getReservaciones() {
        return $this->reservaciones;
    }



}
